core-api-library/library/src/main/java/net/gini/android/core/api/authorization/KAnonymousSessionManager.kt

```diff
@@ -1,22 +1,119 @@
 package net.gini.android.core.api.authorization
 
-import kotlinx.coroutines.flow.Flow
+import net.gini.android.core.api.Resource
+import net.gini.android.core.api.Utils
+import net.gini.android.core.api.authorization.apimodels.SessionToken
 import net.gini.android.core.api.authorization.apimodels.UserRequestModel
-import net.gini.android.core.api.authorization.apimodels.UserResponseModel
+import okhttp3.ResponseBody
+import org.json.JSONObject
+import java.util.*
 
 class KAnonymousSessionManager(
-    userRepository: UserRepository,
-    credentialsStore: CredentialsStore,
-    emailDomain: String,
-    currentSession: Session,
+    private val userRepository: UserRepository,
+    private val credentialsStore: CredentialsStore,
+    private val emailDomain: String
 ): KSessionManager {
-    val userRepository: UserRepository = userRepository
+    private var currentSession: SessionToken? = null
 
-    override suspend fun getSession(): Session {
-        
+    override suspend fun getSession(): SessionToken? {
+        if (currentSession != null && !currentSession!!.hasExpired()) {
+            return currentSession
+        }
+
+        val userCredentials = credentialsStore.userCredentials
+        if (userCredentials == null) {
+            var createResponse = createUser()
+            if (createResponse is Resource.Success) {
+                when (val loginResponse = loginUser()) {
+                    is Resource.Success -> {
+                        currentSession = loginResponse.data
+                        return currentSession
+                    }
+
+                    is Resource.Error -> {
+                        if (isInvalidUserError(loginResponse)) {
+                            currentSession = null
+                            credentialsStore.deleteUserCredentials()
+                            createResponse = createUser()
+                            if (createResponse is Resource.Success) {
+                                currentSession = loginUser().data
+                            }
+                        }
+                    }
+
+                    is Resource.Cancelled -> {
+                        currentSession = null
+                    }
+                }
+            }
+        }
```

Comment:
Else branch is not handled for the `userCredentials == null` if condition -> user won't be logged in if the current session expires or is reset to null.